src/main/scala/coreplex/RTC.scala

```diff
@@ -3,21 +3,18 @@
 package freechips.rocketchip.coreplex
 
 import Chisel._
-import freechips.rocketchip.config.Field
-import freechips.rocketchip.diplomacy.LazyMultiIOModuleImp
+import freechips.rocketchip.diplomacy.{LazyMultiIOModuleImp, DTSTimebase}
 import freechips.rocketchip.devices.tilelink.HasPeripheryClint
 
-/** Real-time clock is based on RTCPeriod relative to system clock.
-  */
-case object RTCPeriod extends Field[Option[Int]]
-
 trait HasRTCModuleImp extends LazyMultiIOModuleImp {
   val outer: HasPeripheryClint
-  private val internalPeriod: Option[Int] = outer.p(RTCPeriod)
-  require(internalPeriod.isDefined, "RTCPeriod is not defined")
+  private val pbusFreq = outer.p(PeripheryBusParams).frequency
+  private val rtcFreq = outer.p(DTSTimebase)
+  require(pbusFreq >= rtcFreq)
+  private val internalPeriod: BigInt = pbusFreq / rtcFreq
```

Comment:
I'd add a check here that the integer division is within 5% of the real division